yarn-project/pxe/src/note_processor/utils/produce_note_daos_for_key.ts

```diff
@@ -0,0 +1,154 @@
+      // Both error messages above occur only when we are dealing with a partial note and are thrown when calling
+      // `note.compute_note_hash()` or `note.compute_nullifier_without_context()`
+      // in `compute_note_hash_and_optionally_a_nullifier` function. It occurs with partial notes because in the
+      // partial flow we receive a note log of a note that is missing some fields here and then we try to compute
+      // the note hash with MSM while some of the fields are zeroed out (or get a nsk for zero npk_m_hash).
+      noteDao = await handlePartialNote(
+        simulator,
+        db,
+        pkM,
+        payload,
+        txHash,
+        noteHashes,
+        dataStartIndexForTx,
+        excludedIndices,
+        logger,
+        unencryptedLogs,
+        daoConstructor,
+      );
+    } else {
+      logger.error(`Could not process note because of "${e}". Discarding note...`);
+    }
+  }
+
+  return [noteDao, deferredNoteDao];
+}
+
+async function handlePartialNote<T>(
+  simulator: AcirSimulator,
+  db: PxeDatabase,
+  pkM: PublicKey,
+  payload: L1NotePayload,
+  txHash: TxHash,
+  noteHashes: Fr[],
+  dataStartIndexForTx: number,
+  excludedIndices: Set<number>,
+  logger: Logger,
+  unencryptedLogs: UnencryptedTxL2Logs,
+  daoConstructor: (payload: L1NotePayload, noteInfo: NoteInfo, dataStartIndexForTx: number, pkM: PublicKey) => T,
+): Promise<T | undefined> {
+  let noteDao: T | undefined;
+
+  for (const functionLogs of unencryptedLogs.functionLogs) {
+    for (const log of functionLogs.logs) {
+      const { data } = log;
+      // It is the expectation that partial notes will have the corresponding unencrypted log be multiple
+      // of Fr.SIZE_IN_BYTES as the nullable fields should be simply concatenated.
+      if (data.length % Fr.SIZE_IN_BYTES === 0) {
+        const nullableFields = [];
+        for (let i = 0; i < data.length; i += Fr.SIZE_IN_BYTES) {
+          const chunk = data.subarray(i, i + Fr.SIZE_IN_BYTES);
+          nullableFields.push(Fr.fromBuffer(chunk));
+        }
+
+        // We insert the nullable fields into the note and then we try to produce the note dao again
+        const payloadWithNullableFields = await addNullableFieldsToPayload(db, payload, nullableFields);
+
+        let deferredNoteDao: DeferredNoteDao | undefined;
+        try {
+          [noteDao, deferredNoteDao] = await produceNoteDaosForKey(
+            simulator,
+            db,
+            pkM,
+            payloadWithNullableFields,
+            txHash,
+            noteHashes,
+            dataStartIndexForTx,
+            excludedIndices,
+            logger,
+            UnencryptedTxL2Logs.empty(), // We set unencrypted logs to empty to prevent infinite recursion.
+            daoConstructor,
+          );
+        } catch (e) {
+          if (!(e as any).message.includes('Could not find key prefix.')) {
+            throw e;
```

Comment:
Why is this error ignored?

Reply:
It's thrown when a different field is incorrectly inserted in place of npk_m_hash when brute forcing the logs. Will add there a comment 👍

Reply:
Added the comment in https://github.com/AztecProtocol/aztec-packages/pull/9009